### DISCUSSION TITLE
Data Collect Interval

### DISCUSSION THREAD

**Discussion by jscholtes** [s-allius/tsun-gen3-proxy#35] [Q&A] [answered]
Hi, erst einmal vielen Dank für das tolle Projekt. 
Es hat auf anhieb funtioniert, meinen MS-800 in mein bestehendes Home Assistant einzubinden :)

In der MQTT Info sehe ich beim Controller einen Parameter "Data Collect Interval"
Das ist aber nicht das Interval, mit dem der Inverter Werte speichert oder? Ich sehe in der payload nur alle 5min einen Wert zb für die output Power

vg
Jürgen

**Comment by s-allius** [comment, marked as answer]
Hallo Jürgen,

schön, dass das Projekt bei dir auf anhieb funktioniert.

Mit `Data Collect Interval` ist das Interval gemeint, mit dem der Wechselrichter die Messdaten intern erfasst. Diese werden dann aggregiert und mit dem `Data Up Interval`, also alle 300s, versendet. Daher bekommt der Proxy auch nur alle 300s Daten, die er dann An den MQTT Broker weiter schickt. Wenn sich Werte nicht ändern, werden diese auch nicht nochmal an den MQTT Broker gesendet. So bleiben die Datenbanken im Home Assistant etwas kleiner.

Gruß Stefan

**Reply by n0rdlicht90** [reply]
Hallo,

auch von mir ein Danke für das tolle Projekt! 
Ist es möglich die Werte für Data Collect- und Data Up- Intervall zu ändern? Ich benötige schnellere Werte für eine Nulleinspeisung. 

Viele Grüße

**Reply by s-allius** [reply]
Prinzipiell ja, aber ich habe die Erfahrung gemacht, dass die Inverser Fw, bei zu schnellen Abfragen Probleme bekommt. Die Frage ist dann aber auch, was du mit den Werten machen möchtest. Die Inverter umzuprogrammieren ist gefährlich, da du dir das interne Flash kaputt schreibst. Diese habe begrenzte Schreibzyklen. TSUN macht dies mit einem UDP Broadcast Paket, welches direkt den Inverser steuert und konfiguriert da nichts um.
